Formula/shapeit4.rb

```diff
@@ -0,0 +1,24 @@
+class Shapeit4 < Formula
+  desc "Segmented HAPlotype Estimation and Imputation Tool"
+  homepage "https://odelaneau.github.io/shapeit4/"
+  url "https://github.com/odelaneau/shapeit4/archive/refs/tags/v4.2.2.tar.gz"
+  sha256 "9f109e307b5cc22ab68e7bf77de2429a9bbb2212d66303386e6a3dd81a5bc556"
+  license "MIT"
+
+  depends_on "boost"
+  depends_on "htslib"
+
+  def install
+    system "sed -i'' -e 's/setiosflags/std::setiosflags/g' src/utils/string_utils.h"
+    system "sed -i'' -e 's/setiosflags/std::setiosflags/g' tools/bingraphsample/src/utils/string_utils.h"
+
+    system "make HTSLIB_LIB=\"-lhts\" BOOST_LIB_IO=\"-lboost_iostreams\" \
+            BOOST_LIB_PO=\"-lboost_program_options\" DYN_LIBS=\"\""
+    mkdir(bin)
+    cp("bin/shapeit4.2", "#{bin}/shapeit4")
+  end
+
+  test do
+    system "#{bin}/shapeit4", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.